integration-test/helpers/test-context.js

```diff
@@ -47,6 +47,11 @@ export function testContext (test) {
                     dataDir,
                     launchOptions
                 )
+
+                // don't allow tests to run until the background page is ready
+                if (context.backgroundPages().length === 0) {
+                    await new Promise((resolve) => context.on('backgroundpage', resolve))
+                }
```

Comment:
this solves the flakiness with the extension tests. Sometimes the background page is already spawn, sometimes not. This ensures we never proceed with it